native_toolchain_rust/lib/src/builder.dart

```diff
@@ -132,7 +137,12 @@ class RustBuilder {
   final String? assetName;
 
   /// Path to the Rust crate directory relative to the package root.
-  final String cratePath;
+  final String? cratePath;
+
+  /// Name of the crate to be built.
+  final String? crateName;
+
+  final CrateType? crateType;
```

Comment:
These need some docs - just wanted to check if I am on the right path first, if yes I will go through this and clean it up.